### ISSUE TITLE
module 'lightkurve' has no attribute 'DesignMatrix'

### ISSUE THREAD

**Issue by nespinoza** [christinahedges/TESS-SIP#2]
Hi @christinahedges!

 Thanks for this wonderful tool! For some reason, however, I can't get this to work. I applied the exact same code as the one on the README:

```
 from tess_sip import SIP
 import lightkurve as lk 
 # Download target pixel files
 tpfs = lk.search_targetpixelfile('TIC 288735205', mission='tess').download_all() 
 # Run SIP
 r = SIP(tpfs)
```

But I get:

```
----> 4 r = SIP(tpfs)

~/.local/lib/python3.7/site-packages/tess_sip/tess_sip.py in SIP(tpfs, sigma, min_period, max_period, nperiods, npca_components, aperture_threshold, sff, sff_kwargs)
    101     with warnings.catch_warnings():
    102         warnings.simplefilter('ignore')
--> 103         bkgs = [lk.DesignMatrix(tpf.flux.value[:, bkg_aper], name='bkg').pca(npca_components).append_constant().to_sparse() for tpf, bkg_aper in zip(tpfs_uncorr, bkg_apers)]
    104         for bkg in bkgs:
    105             bkg.prior_mu[-1] = 1

~/.local/lib/python3.7/site-packages/tess_sip/tess_sip.py in <listcomp>(.0)
    101     with warnings.catch_warnings():
    102         warnings.simplefilter('ignore')
--> 103         bkgs = [lk.DesignMatrix(tpf.flux.value[:, bkg_aper], name='bkg').pca(npca_components).append_constant().to_sparse() for tpf, bkg_aper in zip(tpfs_uncorr, bkg_apers)]
    104         for bkg in bkgs:
    105             bkg.prior_mu[-1] = 1

AttributeError: module 'lightkurve' has no attribute 'DesignMatrix'
```

I'm using the latest version of `lightkurve` (2.0.2) --- also tried the current `dev` version but no dice. Is there a particular version of `lightkurve` that has the `DesignMatrix` attribute?

**EDIT**: Here are some interesting findings. Bottom line: this seems to be a `lightkurve` issue; but even if that's fixed, `TESS-SIP` might need to be modified:

- I checked, and if I revert `lighkurve` to 1.11.3, `DesignMatrix` is there. Since version 2, seems to not be there --- so seems to be a `lightkurve` problem being drawn over here. 
- Interestingly, in the latest `lightkurve` version, there seems to not be an easy way to access `DesignMatrix` --- it seems to be [buried in `correctors`](https://github.com/lightkurve/lightkurve/blob/main/src/lightkurve/correctors/designmatrix.py), but the `correctors` are not loaded in the [`__init__.py` in lightkurve](https://github.com/lightkurve/lightkurve/blob/main/src/lightkurve/__init__.py), so you can't access them (at least not straightforwardly)! I'm going to go on a limb here and ping @barentsen, maybe he has some thoughts on this.
- It seems I can't use TESS-SIP even if I revert back to `lightkurve` 1.11.3. There, the `tpf.flux_bkg` were supposed to be `numpy.ndarray`, so this breaks your code as well when you make `tpf.flux_bkg.value` calls. 

My --- probably very complex --- way of making this work was:
1. Download `lightkurve`'s source code. Add `correctors` to the list of imported sub-packaces in its `__init__.py` --- install this "hacked" version of `lightkurve`.
2. Change all the `lk.DesignMatrix` and `lk.SparseDesignMatrix` calls to `lk.correctors.DesignMatrix` and `lk.correctors.SparseDesignMatrix` in the `tess_sip.py` script.

After this I still got a problem on [line 120 and 123](https://github.com/christinahedges/TESS-SIP/blob/b6e05889560d27b493397c669ca7d2b02c8be94c/src/tess_sip/tess_sip.py#L123) --- `dm.prior_sigma` is a `numpy.ndarray`, so does not have a `value` attribute. Fixing that I can use `tess-sip`.

**Comment by barentsen** [comment]
As of Lightkurve v2.0.0, the following statement:
```python
from lightkurve import DesignMatrix
```
needs to be replaced by
```python
from lightkurve.correctors import DesignMatrix
```

This change was made to avoid having to load the correctors sub-package and all its dependencies on each call to `import lightkurve`.  We likely underestimated the impact of this change on existing packages however (e.g. eleanor, cf. https://github.com/lightkurve/lightkurve/issues/975).

I will release Lightkurve v2.0.3 later today to fix this issue.

**Comment by nespinoza** [comment]
Thanks @barentsen for the prompt reply! That works and makes sense --- I'll make a PR to `TESS-SIP` to add this myself, given I've already gone through the work of doing this :-).

**Comment by barentsen** [comment]
Lightkurve v2.0.3 has now been released.

This means `lk.DesignMatrix` and similar classes from the correctors sub-package can be imported again in the same way as in Lightkurve v1.

Thank you for drawing my attention to this issue @nespinoza! 👍

**Comment by nespinoza** [comment]
Awesome, thanks @barentsen! 

@christinahedges: this solves the issue with the `lk.DesignMatrix` call --- but still does not solve the issue on line 120 and 123 mentioned above. PR #3 makes that small change as well.

**Comment by christinahedges** [comment]
Hi @nespinoza, I believe this has been fixed in [the updated v2.0.4 of `lightkurve`](https://github.com/lightkurve/lightkurve/releases/tag/v2.0.4) and [the new v1.0.7 of `tess-sip`](https://github.com/christinahedges/TESS-SIP/releases/tag/1.0.7)! Thank you for highlighting this and contributing a fix, I'm going to close this issue now!
